xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -802,7 +821,6 @@ void onError(Status error) {
     }
 
     private void notifyWatcher(ResourceWatcher watcher, ResourceUpdate update) {
-      // TODO(chengyuanzhang): should invoke callbacks with watcher's own executor.
```

Comment:
Why is this comment removed but not the one on line 818 for `onError`?

Reply:
Discussed with Eric, it's ok to run it with XdsClient's channel thread.

Reply:
Understood, but you have the same comment on line 818 for `onError`. So it is okay to run `notifyWatcher` on the XdsClient channel thread but you are planning to move the onError callback on the caller provided executor? I think line 818 should be removed as well

Reply:
Oops, missed to delete that one. Thanks for catching it.